Code/Mods/UI/Camera.cs

```diff
@@ -285,9 +299,44 @@ private static void Character_SetZoomMode_Post(Character __instance, ref bool _z
         if (!_perPlayerSettings[player.ID]._toggle)
             return;
         #endregion
-
+        
         __instance.CharacterCamera.ZoomSensModifier = 1f;
-        _perPlayerSettings[player.ID].StartAimCoroutine(__instance, _zoomed);
+        if (_zoomed)
+        {
+            _perPlayerSettings[player.ID]._cachedZoom = _perPlayerSettings[player.ID]._zoomControlAmount.Value;
+            _perPlayerSettings[player.ID].StartAimCoroutine(__instance, _zoomed);
+        }
+
+    }
+
+    [HarmonyPrefix, HarmonyPatch(typeof(CharacterCamera), nameof(CharacterCamera.SetCameraMode))]
+    private static void CharacterCamera_SetCameraMode_Pre(CharacterCamera __instance, ref CharacterCamera.CameraModes _mode)
+    {
+        bool CameraGoingUnzoomed = __instance.m_currentViewMode == CharacterCamera.CameraModes.Zoom && _mode != CharacterCamera.CameraModes.Zoom;
+        if (CameraGoingUnzoomed)
+        {
+            Players.TryGetLocal(__instance, out Players.Data player);
+            bool playerValidTarget = player != null && _perPlayerSettings[player.ID]._toggle;
+            bool playerLeavingZoomUnset = !_perPlayerSettings[player.ID]._leavingZoom;
+            if (playerValidTarget && playerLeavingZoomUnset)
+            {
+                _perPlayerSettings[player.ID]._leavingZoom = true;
+            }
+        }
+    }
+
+    [HarmonyPostfix, HarmonyPatch(typeof(CharacterCamera), nameof(CharacterCamera.SetCameraMode))]
+    private static void CharacterCamera_SetCameraMode_Post(CharacterCamera __instance, ref CharacterCamera.CameraModes _mode)
+    {
+
+        Players.TryGetLocal(__instance, out Players.Data player);
+        bool playerValidTarget = player != null && _perPlayerSettings[player.ID]._toggle;
+        bool playerLeavingZoomSet = _perPlayerSettings[player.ID]._leavingZoom;
+        if (playerValidTarget && playerLeavingZoomSet)
+        {
+            _perPlayerSettings[player.ID].StartAimCoroutine(player.Character, false);
+            _perPlayerSettings[player.ID]._leavingZoom = false;
+        }
```

Comment:
Also look good. Here I would not have an early exit as that would make the code less readable

Reply:
Does that even matter if _leavingZoom isn't used by the coroutine?